File: src/vibe_kanban_clone/schemas/task.py
"""Task schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class TaskCreate(BaseModel):
    """Schema for creating a task."""

    project_id: int
    column_id: int | None = None
    title: str = Field(min_length=1, max_length=255)
    description: str | None = Field(default=None, min_length=1, max_length=5000)
    priority: int = Field(default=4, ge=1, le=5)
    assigned_agent_id: int | None = None


class TaskRead(BaseModel):
    """Schema for reading a task."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: int
    column_id: int | None
    title: str
    description: str | None
    priority: int
    result: str | None
    assigned_agent_id: int | None
    version: int
    claimed_at: datetime | None
    created_at: datetime
    updated_at: datetime


class TaskUpdate(BaseModel):
    """Schema for updating a task."""

    title: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = Field(default=None, min_length=1, max_length=5000)
    priority: int | None = Field(default=None, ge=1, le=5)
    result: str | None = Field(default=None, min_length=1, max_length=50000)
    assigned_agent_id: int | None = None
    expected_version: int | None = None


class TaskMove(BaseModel):
    """Schema for moving a task to another column."""

    column_id: int
    expected_version: int | None = None


class TaskRunCreate(BaseModel):
    """Schema for starting a task execution run."""

    model_id: int
    prompt: str | None = None

File: server/src/schemas/task.ts
import { z } from "zod";

export const TaskCreate = z.object({
  project_id: z.number(),
  column_id: z.number().nullable().optional(),
  title: z.string().min(1).max(255),
  description: z.string().min(1).max(5000).optional(),
  priority: z.number().min(1).max(5).default(4),
  assigned_agent_id: z.number().nullable().optional(),
});

export const TaskRead = z.object({
  id: z.number(),
  project_id: z.number(),
  column_id: z.number().nullable(),
  title: z.string(),
  description: z.string().nullable(),
  priority: z.number(),
  result: z.string().nullable(),
  assigned_agent_id: z.number().nullable(),
  version: z.number(),
  claimed_at: z.string().or(z.date()).nullable(),
  created_at: z.string().or(z.date()),
  updated_at: z.string().or(z.date()),
});

export const TaskUpdate = z.object({
  title: z.string().min(1).max(255).optional(),
  description: z.string().min(1).max(5000).optional(),
  priority: z.number().min(1).max(5).optional(),
  result: z.string().min(1).max(50000).optional(),
  assigned_agent_id: z.number().nullable().optional(),
  expected_version: z.number().optional(),
});

export const TaskMove = z.object({
  column_id: z.number(),
  expected_version: z.number().optional(),
});

export const TaskRunCreate = z.object({
  model_id: z.number(),
  prompt: z.string().optional(),
});

export type TaskCreate = z.infer<typeof TaskCreate>;
export type TaskRead = z.infer<typeof TaskRead>;
export type TaskUpdate = z.infer<typeof TaskUpdate>;
export type TaskMove = z.infer<typeof TaskMove>;
export type TaskRunCreate = z.infer<typeof TaskRunCreate>;
